skare3_tools/packages.py

```diff
@@ -167,21 +167,18 @@ def wrapper(*args, update=False, **kwargs):
             if update_policy is not None and result is not None:
                 update = update or update_policy(filename, result)
             if not dir_access_ok(filename):
-                if result is None:
-                    raise Exception(
```

Comment:
Is this removed because the exception has changed or?

Reply:
Ah, I had not noticed this was in this branch. This is a change unrelated to the promotion script. This is used 

Now it raises no exception. Before this change, if the user had no **write** access to the cache file, it would raise an exception, but this is not needed. I changed the logic so it would always tries to read. At this point it only checks access, and uses that information below (right before writing).

It can still raise an exception if the cache file exists and it has no read access.

I might just edit the PR description to make it a larger PR instead of rebasing. I'll check.

Reply:
Gotcha.  Fine with an unrelated fix if in the description - though might want to add a note about testing this.